### PR TITLE
WIP: Expand with base

### PR DESCRIPTION
**Pull request by fredbi**

### CODE REVIEW

File: helpers.go [hunk 1]
```diff
@@ -177,37 +180,46 @@ func (h *paramHelper) safeExpandedParamsFor(path, method, operationID string, re
 }
 
 func (h *paramHelper) resolveParam(path, method, operationID string, param *spec.Parameter, s *SpecValidator) (*spec.Parameter, *Result) {
-	// Expand parameter with $ref if needed
+	// Ensure parameter is expanded
+	var err error
 	res := new(Result)
+	isRef := param.Ref.String() != ""
+	if s.spec.SpecFilePath() == "" {
+		err = spec.ExpandParameterWithRoot(param, s.spec.Spec(), nil)
```
Comment (golangci) [inline helpers.go]: ExpandParameterWithRoot not declared by package spec

File: helpers.go [hunk 2]
```diff
@@ -218,21 +230,29 @@ type responseHelper struct {
...
-	if response.Ref.String() != "" {
-		err := spec.ExpandResponse(response, s.spec.SpecFilePath())
-		if err != nil { // Safeguard
-			// NOTE: we may enter here when the whole response is an unresolved $ref.
-			errorHelp.addPointerError(res, err, response.Ref.String(), path)
-			return nil, res
-		}
+	if s.spec.SpecFilePath() == "" {
+		// there is no physical document to resolve $ref in response
+		err = spec.ExpandResponseWithRoot(response, s.spec.Spec(), nil)
```
Comment (golangci) [inline helpers.go]: ExpandResponseWithRoot not declared by package spec

**Comment by fredbi** [comment]
Awaiting PR to be merged in spec repo:
- https://github.com/go-openapi/spec/pull/83

File: .golangci.yml [hunk 3]
```diff
@@ -14,6 +14,8 @@ linters-settings:
     min-occurrences: 2
 
 linters:
-  enable-all: true
+  #enable-all: true
+  disable-all: true
```
Comment (casualjim) [inline .golangci.yml]: disable all or enable all?

Comment (fredbi) [reply .golangci.yml]: just temporary setup : golangci does not tolerate when importing packages from another branch. I will revert this to original, now that spec is merged.

Comment (fredbi) [reply .golangci.yml]: I remains an issue I don't  know how to solve: when extending the interface of a dependent pkg and trying to test the integration. I know how to twist travis, etc... but not golangci

**Comment by fredbi** [comment]
Still WIP. There is a pb with the test suite...

**Comment by gregmarr** [comment]
@casualjim FYI, this PR was marked as still a WIP, including @fredbi 's most recent comment.

**Comment by fredbi** [comment]
Yep I was in a middle of a minor fix...
It does not matter much: it is not blocking (improving how successful the UT is: it is currently polluted by spurious warnings when running on CI). I will have to push another time to finish the job...

**Pull request merged**